packages/devtools_app/lib/src/shared/http/http_request_data.dart

```diff
@@ -55,8 +57,124 @@ class DartIOHttpRequestData extends NetworkRequest {
       HttpProfileRequestRef.parse(modifiedRequestData)!,
       requestFullDataFromVmService: false,
     )
-      .._responseBody = responseContent?['text'].toString()
-      .._requestBody = requestPostData?['text'].toString();
+      .._responseBody =
+          responseContent?[HttpRequestDataKeys.text.name].toString()
+      .._requestBody =
+          requestPostData?[HttpRequestDataKeys.text.name].toString();
+  }
+  //TODO go through all parameters, to check if they are correctly added.
+  @override
+  Map<String, Object?> toJson() {
+    try {
+      return {
```

Comment:
I'm surprised this is so large in comparison to the `DartIOHttpRequestData.fromJson` method.  I would have expected serialization logic to have a similar number of keys we have to read from / write. 

Perhaps there is something better we can do here to break this up into smaller pieces. For example, in the `fromJson` method, we are using the existing `HttpProfileRequestRef.parse` method from `package:vm_service`. Perhaps we could add an extension method on the `HttpProfileRequestRef` class to implement the inverse of `HttpProfileRequestRef.parse` as a `toJson` method:

```
extension on HttpProfileRequestRef {
  Map<String, Object?> toJson() { ... }
}
```
This may require implementing other toJson methods for non-primitive fields on the `HttpProfileRequestRef` class, like `HttpProfileRequestData` and `HttpProfileResponseData` for example.

Once implemented, you should easily be able to go back and forth between `HttpProfileRequestRef.parse` and `HttpProfileRequestRef.toJson` without modifying the data. E.g. some test like:
```
final someRequest = HttpProfileRequestRef(...);
final requestJson = someRequest.toJson();
final parsedRequest = HttpProfileRequestRef.parse(requestJson);
// verify that someRequest and parsedRequest have identical content.
```

Then our `DartIOHttpRequestData.toJson` return could look something like:
```
return {
  'request': _request.toJson();
  // any other data from the DartIOHttpRequestData class we need to store, if any
}
```

Reply:
Yes, I understand we should delegate it to respective classes, actually I had added all of it here as toJson wasn't readily available like 'parse', as this class is in external package, but I realise now that we can achieve same using an extension/utility function.